pkg/kudoctl/packages/resolver/resolver_local.go

```diff
@@ -2,50 +2,58 @@ package resolver
 
 import (
 	"fmt"
+	"path/filepath"
 	"strings"
 
 	"github.com/spf13/afero"
 
-	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
 	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
 	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/reader"
 )
 
-// LocalResolver will find local operator packages: folders or tgz
-type LocalResolver struct {
+// LocalHelper will find local operator packages: folders or tgz
+type LocalHelper struct {
 	fs afero.Fs
 }
 
-// Resolve a local package. The path can be relative or absolute location of the packages.
-// Order of the discovery is:
-// 1. tarball
-// 2. dir based
-func (f *LocalResolver) Resolve(name string, appVersion string, operatorVersion string) (*packages.Resources, error) {
-	//	make sure file exists
-	_, err := f.fs.Stat(name)
-	if err != nil {
-		return nil, err
+func (f *LocalHelper) LocalPackagePath(path string) (string, error) {
+	fi, err := f.fs.Stat(path)
+
+	// force local operators usage to be either absolute or express a relative path
+	// or put another way, a name can NOT be mistaken to be the name of a local folder
+	if filepath.Base(path) != path && err == nil {
+		var abs string
+		abs, err = filepath.Abs(path)
+		if err != nil {
+			return "", err
+		}
+
+		// expecting either a folder or .tgz file
+		if fi.IsDir() || (fi.Mode().IsRegular() && strings.HasSuffix(abs, ".tgz")) {
+			return abs, nil
+		}
 	}
+	return "", fmt.Errorf("%s is not a valid local package path", path)
+}
 
-	//	make sure file exists
-	fi, err := f.fs.Stat(name)
-	if err != nil {
-		return nil, err
+func (f *LocalHelper) ResolveTar(out afero.Fs, path string) (*packages.Resources, error) {
```

Comment:
`out` should be a return value, as it's always called with an `afero.NewMemMapFs`.